### ISSUE TITLE
Roundtrip failure for double -0x1.e42427b42cb42p+949

### ISSUE THREAD

**Issue by pauldreik** [stephenberry/glaze#1183]
According to the FAQ, numbers should be roundtrippable:
https://github.com/stephenberry/glaze/blob/main/docs/FAQ.md#is-the-writeread-api-deterministic

Here is an example which seems to be 1 ULP off for all three numbers:
```cpp

struct D { double value;};

"double roundtrip"_test = [] {
       for (const double expected:{
            -0x1.e42427b42cb42p+949,
           -0x1.3ffff0d0ddb37p+725,
           0x1.73d40c08b20ffp-395
   }) {
           const D d{expected};
           auto str = glz::write_json(d).value_or(std::string{});
           auto restored = glz::read_json<D>(str);
           expect(restored.has_value());
           expect(restored.value().value == d.value);

       }
   };
```

The first value is https://float.exposed/0xfb4e42427b42cb42 and does not seem to be extreme (close to subnormal, max, min etc).

Side question: I could not put struct D inside the function (to narrow the scope where it is visible). Is that expected?

**Comment by stephenberry** [comment]
> Side question: I could not put struct D inside the function (to narrow the scope where it is visible). Is that expected?

Yes, this is expected behavior with the unit test framework and the current pure reflection mechanisms. It is not a limitation typically encountered in software.

**Comment by stephenberry** [comment]
These 1 ULP errors have been fixed with the merge of #1192
